google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/read_rows/RowMerger.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub.read_rows;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.ReadRowsResponse;
+import com.google.cloud.bigtable.data.v2.wrappers.RowAdapter.RowBuilder;
+import com.google.cloud.bigtable.gaxx.reframing.Reframer;
+import com.google.common.base.Preconditions;
+
+/**
+ * An implementation of a {@link Reframer} that feeds the row merging {@link StateMachine}.
+ *
+ * <p>{@link com.google.cloud.bigtable.gaxx.reframing.ReframingResponseObserver} pushes {@link
+ * ReadRowsResponse.CellChunk}s into this class and pops fully merged logical rows.
+ *
+ * <p>This class is considered an internal implementation detail and not meant to be used by
+ * applications.
+ *
+ * <p>Package-private for internal use.
+ */
+@InternalApi
+class RowMerger<RowT> implements Reframer<RowT, ReadRowsResponse> {
```

Comment:
For the sake of understanding our internal code, could you provide an example on how this class should be used?

Reply:
Would adding a link to https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/reframing/ReframingResponseObserver.java be enough?

Reply:
I prefer having an example here. I don't see an example in the interface, beside, we may used RowMerger differently.

Reply:
added an example